service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/EhrAcknowledgementTest.java

```diff
@@ -0,0 +1,111 @@
+import uk.nhs.adaptors.gp2gp.testcontainers.ActiveMQExtension;
+import uk.nhs.adaptors.gp2gp.testcontainers.MongoDBExtension;
+
+@ExtendWith({SpringExtension.class, MongoDBExtension.class, ActiveMQExtension.class})
+@SpringBootTest
+@DirtiesContext
+public class EhrAcknowledgementTest {
+    private static final String ROOT_ID = "75049C80-5271-11EA-9384-E83935108FD5";
+    private static final String MESSAGE_REF = "BA229526-ADC6-4F1D-970E-CD9E78A6830E";
+    private static final String BUSINESS_ERROR_CODE = "99";
+    private static final String BUSINESS_ERROR_DISPLAY = "Unexpected condition";
+    private static final String REJECTED_ERROR_CODE = "18";
+    private static final String REJECTED_ERROR_DISPLAY = "Request message not well-formed or not able to be processed";
+
+    @Autowired
+    private EhrExtractRequestHandler ehrExtractRequestHandler;
+    @Autowired
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+    @Autowired
+    private XPathService xPathService;
+    @Value("classpath:inbound/acknowledgement/MCCI_IN010000UK13_body_AA.xml")
+    private Resource positiveAck;
+    @Value("classpath:inbound/acknowledgement/MCCI_IN010000UK13_body_AR.xml")
+    private Resource rejectedAck;
+    @Value("classpath:inbound/acknowledgement/MCCI_IN010000UK13_body_AE.xml")
+    private Resource businessErrorAck;
+
+    @SneakyThrows
+    @Test
+    public void When_EhrReceivedAckAA_Expect_DbUpdatedAndNoErrors() {
+        Optional<EhrExtractStatus> ehrExtract = setUpAndHandleAck(positiveAck);
+
+        assertThat(ehrExtract).isPresent()
+            .map(EhrExtractStatus::getEhrReceivedAcknowledgement)
+            .hasValueSatisfying(ack -> assertAll(
+                () -> assertThat(ack.getReceived()).isEqualTo(ack.getConversationClosed()),
+                () -> assertThat(ack.getMessageRef()).isEqualTo(MESSAGE_REF),
+                () -> assertThat(ack.getRootId()).isEqualTo(ROOT_ID),
+                () -> assertThat(ack.getErrors()).isNull()
+            ));
+    }
+
+    @SneakyThrows
+    @Test
+    public void When_EhrReceivedNackAE_Expect_DbUpdatedWithErrors() {
+        Optional<EhrExtractStatus> ehrExtract = setUpAndHandleAck(businessErrorAck);
+
+        assertThat(ehrExtract).isPresent()
+            .map(EhrExtractStatus::getEhrReceivedAcknowledgement)
+            .hasValueSatisfying(ack -> assertAll(
+                () -> assertThat(ack.getReceived()).isNotNull(),
+                () -> assertThat(ack.getConversationClosed()).isNull(),
+                () -> assertThat(ack.getMessageRef()).isEqualTo(MESSAGE_REF),
+                () -> assertThat(ack.getRootId()).isEqualTo(ROOT_ID),
+                () -> assertThat(ack.getErrors()).isNotEmpty(),
+                () -> assertThat(ack.getErrors().get(0).getCode()).isEqualTo(BUSINESS_ERROR_CODE),
+                () -> assertThat(ack.getErrors().get(0).getDisplay()).isEqualTo(BUSINESS_ERROR_DISPLAY)
```

Comment:
These can also be combined, if you think it improves the semantics of the assertions
```suggestion
                () -> assertThat(ack.getErrors()).isNotEmpty()
                    .first().satisfies(error -> assertAll(
                        () -> assertThat(error.getCode()).isEqualTo(BUSINESS_ERROR_CODE),
                        () -> assertThat(error.getDisplay()).isEqualTo(BUSINESS_ERROR_DISPLAY)))
```

Reply:
Obviously this style of assertion can quickly take you down a rabbit-hole, which we saw a bit on P&S, but for me this is still okay.